pkg/config/config.go

```diff
@@ -161,9 +166,28 @@ func (gc *Config) configureProxies(httpProxy, httpsProxy, noProxy string) {
 	}
 
 	if noProxy != "" {
+		noProxy, err := addKubernetesServiceHostInNoProxy(noProxy)
+		if err != nil {
+			return err
+		}
 		os.Setenv(noProxyEnvVar, noProxy)
 		os.Setenv(strings.ToUpper(noProxyEnvVar), noProxy)
 	}
+	return nil
+}
+
+func addKubernetesServiceHostInNoProxy(noProxy string) (string, error) {
+	noProxyList := strings.Split(noProxy, ",")
+	for _, noProxyItem := range noProxyList {
```

Comment:
this loop seems a bit weird -- we call strings.Replace in case we match (but against a first match which may not even be the matched split item), and we can call it multiple times. on top of that it seems that no_proxy is somewhat weird in various formats https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/ so may be it's not even comma separated.

i would recommend abandoning this logic and just sticking with plain strings.Replace like it was before. if we have concerns of "collision" with actual no proxy info, then lets change name to KAPPCTRL_KUBERNETES_SERVICE_HOST. i dont think we should worry about xyz-KUBE... case.

Reply:
Reverted back and now using `KAPPCTRL_KUBERNETES_SERVICE_HOST`.